packages/engine/Source/DataSources/EntityCluster.js

```diff
@@ -932,6 +957,11 @@ EntityCluster.prototype.update = function (frameState) {
  * </p>
  */
 EntityCluster.prototype.destroy = function () {
+  if (defined(this._removeEventListener)) {
```

Comment:
I moved this first, just in case, to prevent any race conditions with destroyed objects.